### PR TITLE
obsidian: include both light and dark themes when polarity is either

### PR DESCRIPTION
**Pull request by repparw**
When `stylix.polarity` is set to `"either"`, the obsidian module generates a CSS snippet with the selector `.theme-either`, which is not a valid Obsidian theme class. This causes the stylix theme to silently fail to apply in Obsidian.

This change generates CSS for both `.theme-light` and `.theme-dark` when polarity is `"either"`, ensuring the stylix colors apply regardless of whether Obsidian is in light mode, dark mode, or set to adapt to system.

For `"light"` or `"dark"` polarity, behavior is unchanged.

### CODE REVIEW

File: modules/obsidian/hm.nix [hunk 1]
```diff
@@ -32,12 +32,15 @@ mkTarget {
         colors,
         polarity,
       }:
+      let
+        effectivePolarity = if polarity == "either" then "dark" else polarity;
+      in
```
Comment (TheColorman) [inline modules/obsidian/hm.nix:37]: The `"either"` value is new to me, but it seems the scheme is based on whichever generated palette is the closest to the chosen image.

This means it is possible for `"either"` to be a light scheme, in which case the system scheme is set to "light" (or at least Obsidian interprets it as light). This causes Obsidian to display the light theme (as its set to "Adapt to system" by default), which will not show the Stylix theme.

Not really sure what the solution is here other than generating both a dark and light theme when using `"either"`.

Comment (repparw) [reply modules/obsidian/hm.nix:37]: Not sure if it applies with generated palettes, haven't tried. 
Issue happened to me because I set a base16scheme manually (tokyodark-terminal)
and I didn't manually set polarity as it didn't matter to me.

we could set appearance.baseColorScheme with the same criteria (polarity or "dark" if either) instead of adapt to system. (Dunno where adapt to system is set in stylix globally?)

Comment (repparw) [reply modules/obsidian/hm.nix:37]: Generating both seems like the cleaner fix

**Review by TheColorman — APPROVED** [review]
Looks good to me now! Perhaps we should add testbeds for the "either" theme in the near future to properly test edge cases like this?

File: modules/obsidian/hm.nix [hunk 2]
```diff
@@ -32,56 +32,50 @@ mkTarget {
...
+        themeSelectors =
+          if polarity == "either" then
+            [
+              "light"
+              "dark"
+            ]
+          else
+            [ polarity ];
+
+        mkSnippet =
```
Comment (0xda157) [inline modules/obsidian/hm.nix]: "snippet" is pretty much meaningless to someone trying to read this code, maybe consider something like `mkTheme`

**Review by 0xda157 — CHANGES_REQUESTED** [review]

**Review by trueNAHO — CHANGES_REQUESTED** [review]
This PR can be substantially simplified with the following standard pattern:

```diff
diff --git a/modules/obsidian/hm.nix b/modules/obsidian/hm.nix
index 9ed3ab15..baf9ad36 100644
--- a/modules/obsidian/hm.nix
+++ b/modules/obsidian/hm.nix
@@ -32,12 +32,15 @@ mkTarget {
         colors,
         polarity,
       }:
+      let
+        polarity' = if polarity == "dark" then polarity else "light";
+      in
       {
         programs.obsidian.defaultSettings.cssSnippets = with colors.withHashtag; [
           {
             name = "Stylix Config";
             text = ''
-              .theme-${polarity} {
+              .theme-${polarity'} {
                   /* Base Colors */
                   --color-base-00: ${base00};
                   --color-base-05: ${base00};
@@ -63,7 +66,7 @@ mkTarget {
             {
               name = "Stylix Config";
               text = ''
-                .theme-${polarity} {
+                .theme-${polarity'} {
                     /* Base Colors */
                     --color-base-00: ${base00};
                     --color-base-05: ${base00};
```

The `mkTheme` deduplication would be nice but is technically out-of-scope, as it is no longer necessary with my variant.

Although dark and light theme switching in applications is planned with https://github.com/nix-community/stylix/issues/447, relying on the `"either"` variant is brittle because the current plans are to remove it in the near future to avoid ambiguity and confusion.

**Review by trueNAHO — APPROVED** [review]
LGTM. I will merge this, as it should resolve the mentioned issue.

Feel free to resolve the pending cleanup in a follow-up PR if you feel like it:

> The `mkTheme` deduplication would be nice but is technically out-of-scope, as it is no longer necessary with my variant.
